networking/p2p/src/integration-test/java/tech/pegasys/artemis/networking/p2p/jvmlibp2p/JvmLibP2pSmokeIntegrationTest.java

```diff
@@ -45,9 +48,25 @@ public void shouldConnectToPeers() throws Exception {
     network1.connect(network2.getPeerAddress());
     Waiter.waitFor(
         () -> {
-          assertThat(network1.getPeerCount()).isEqualTo(1);
-          assertThat(network2.getPeerCount()).isEqualTo(1);
+          assertThat(network1.getPeerManager().getPeerCount()).isEqualTo(1);
+          assertThat(network2.getPeerManager().getPeerCount()).isEqualTo(1);
         });
+
+    final Peer network2ViewOfPeer1 =
+        network2.getPeerManager().getPeer(network1.getPeerId()).orElseThrow();
+    assertThat(network2ViewOfPeer1.getStatus().getForkVersion()).isEqualTo(Fork.VERSION_ZERO);
+    assertThat(network2ViewOfPeer1.getStatus().getFinalizedRoot()).isEqualTo(Bytes32.ZERO);
+    assertThat(network2ViewOfPeer1.getStatus().getFinalizedEpoch()).isEqualTo(UnsignedLong.ZERO);
+    assertThat(network2ViewOfPeer1.getStatus().getHeadRoot()).isEqualTo(Bytes32.ZERO);
+    assertThat(network2ViewOfPeer1.getStatus().getHeadSlot()).isEqualTo(UnsignedLong.ZERO);
+
+    final Peer network1ViewOfPeer2 =
+        network1.getPeerManager().getPeer(network2.getPeerId()).orElseThrow();
+    assertThat(network1ViewOfPeer2.getStatus().getForkVersion()).isEqualTo(Fork.VERSION_ZERO);
+    assertThat(network1ViewOfPeer2.getStatus().getFinalizedRoot()).isEqualTo(Bytes32.ZERO);
+    assertThat(network1ViewOfPeer2.getStatus().getFinalizedEpoch()).isEqualTo(UnsignedLong.ZERO);
+    assertThat(network1ViewOfPeer2.getStatus().getHeadRoot()).isEqualTo(Bytes32.ZERO);
+    assertThat(network1ViewOfPeer2.getStatus().getHeadSlot()).isEqualTo(UnsignedLong.ZERO);
```

Comment:
That's the next PR - it would currently fail due to a couple of bugs in our Resq/Resp framework.  We are holding  references to a Netty ByteBuf without retain so it get's released underneath us and we're actually always echoing back the request we receive, not the response we generated.